src/resmom/linux/alps.c

```diff
@@ -3187,12 +3189,28 @@ inventory_to_vnodes(basil_response_t *brp)
 			default:
 				continue;
 		}
-		for (seg=node->segments; seg; seg=seg->next) {
-			char	vname[128];
 
-			snprintf(vname, sizeof(vname), "%s_%ld_%d",
-				mpphost, node->node_id, seg->ordinal);
-			vname[sizeof(vname)-1]='\0';
+		totcpus = 0;
+		totmem = 0;
+
+		/* Only do this for nodes that have accelerators. */
+                if (node->accelerators) {
+                        for (accel=node->accelerators, totaccel=0;
+                                accel; accel=accel->next) {
+                                if (accel->state == basil_accel_state_up)
+                                        /* Only count them if the state is UP */
+                                        totaccel++;
+                        }
+                }
```

Comment:
Why was this block moved from its original location? The value of totaccel is not used until the next block that tests for node->accelerators